OpenRA.Mods.Common/Pathfinder/PathGraph.cs

```diff
@@ -73,17 +73,26 @@ public GraphConnection(CPos destination, int cost)
 		}
 	}
 
-	sealed class PathGraph : IGraph<CellInfo>
+	public class PathGraph : IGraph<CellInfo>
 	{
 		public const int CostForInvalidCell = int.MaxValue;
 
-		public Actor Actor { get; private set; }
-		public World World { get; private set; }
-		public Func<CPos, bool> CustomBlock { get; set; }
-		public Func<CPos, int> CustomCost { get; set; }
-		public int LaneBias { get; set; }
-		public bool InReverse { get; set; }
-		public Actor IgnoreActor { get; set; }
+		public Actor Actor { get { return actor; } }
+		public World World { get { return world; } }
+		public Func<CPos, bool> CustomBlock { get { return customBlock; } set { customBlock = value; } }
+		public Func<CPos, int> CustomCost { get { return customCost; } set { customCost = value; } }
+		public int LaneBias { get { return laneBias; } set { laneBias = value; } }
+		public bool InReverse { get { return inReverse; } set { inReverse = value; } }
+		public Actor IgnoreActor { get { return ignoreActor; } set { ignoreActor = value; } }
+
+		// PERF: Use fields over virtual properties
+		readonly Actor actor;
+		readonly World world;
+		Func<CPos, bool> customBlock;
+		Func<CPos, int> customCost;
+		int laneBias;
+		bool inReverse;
+		Actor ignoreActor;
```

Comment:
Also in class `class D : C'? Also when there is a setter? Because with a setter i can imagine it no longer 'dares' to?
I read somewhere C# wouldn't/couldn't. Good to know.

Reply:
In both those cases its able to inline it proper.

Reply:
So shall I, some time later, remove both interfaces and change the fields back to properties? Or only the last.

Reply:
Both.